level1c4pps/vgac2pps_lib.py

```diff
@@ -343,17 +473,78 @@ def convert_to_noaa19(scene, sbaf_version):
             filt = filt & (scene["sunzenith"].values >= scaling["min_sunzenith"])
         if "max_sunzenith" in scaling:
             filt = filt & (scene["sunzenith"].values < scaling["max_sunzenith"])
-        scene[viirs_channel].values = np.where(
-            filt,
-            slope * scene[viirs_channel].values + offset,
-            scene[viirs_channel].values
-        )
-        logger.info(f"{avhhr_chan:<13} = {slope:<6}*{viirs_channel:<3}+{offset:<5} ({comment})")
+
+        if slope is not None:
+            # Then simple linear regression
+            scene[viirs_channel].values = np.where(
+                filt,
+                slope * scene[viirs_channel].values + offset,
+                scene[viirs_channel].values
+                )
+            logger.info(f"{avhrr_chan:<13} = {slope:<6}*{viirs_channel:<3}+{offset:<5} ({comment})")
+        else:
+            if avhrr_chan == "tb37_twilight":
+                # 70 < SZA < 85: BT = (1-f)*BT(day) + f*BT(night), f=(SZA-70)/15
+
+                f = (scene["sunzenith"].values - scaling["min_sunzenith"])/15
+                tb37_day_slope = SBAF[sbaf_version]["tb37_day"]["slope"]
+                tb37_day_offset = SBAF[sbaf_version]["tb37_day"]["offset"]
+                tb37_night_slope = SBAF[sbaf_version]["tb37_night"]["slope"]
+                tb37_night_offset = SBAF[sbaf_version]["tb37_night"]["offset"]
+                tb37_day = tb37_day_slope * scene[viirs_channel].values + tb37_day_offset
+                tb37_night = tb37_night_slope * scene[viirs_channel].values + tb37_night_offset
+
+                scene[viirs_channel].values = np.where(
+                    filt,
+                    (1-f)*tb37_day + f*tb37_night,
+                    scene[viirs_channel].values
+                )
+
+            elif avhrr_chan == "tb12":
+                # BT(5) = BT(ch4)-1.1646*(BT(M15)-BT(M16))-0.235
+                scene[viirs_channel].values = scene["M15"].values-1.1646*(tb11_original-scene["M16"].values)-0.235
+            else:
+                logger.exception(f'Unknown channel, {avhrr_chan}, or missing slope parameter')
+            logger.info(f"{avhrr_chan:<13}: ({comment})")
+
+
+def convert_to_noaa19(scene, sbaf_version):
+    """ Applies AVHRR SBAF to VGAC channels"""
+
+    logger.info(f"Using SBAF_{sbaf_version}")
+
+    if "NN" not in sbaf_version:
+        if sbaf_version == "KNMI_v2":
+            convert_to_noaa19_KNMI_v2(scene, sbaf_version)
+        else:
+            # 1 channel linear regression
```

Comment:
would be nicer if this was specified in a separate function, to make it "symmetric" with the other options.